src/test/ui/closures/2229_closure_analysis/migrations/mir_calls_to_shims.rs

```diff
@@ -10,12 +10,15 @@ use std::panic;
 
 fn foo_diverges() -> ! { panic!() }
 
+struct Foo(i32);
+
 fn assert_panics<F>(f: F) where F: FnOnce() {
+    let f = Foo(0);
     let f = panic::AssertUnwindSafe(f);
     let result = panic::catch_unwind(move || {
         //~^ ERROR: 'Sync' trait implementation affected for closure because of `capture_disjoint_fields`
         //~| HELP: add a dummy let to cause `f` to be fully captured
-        f.0()
+        f.0.0
```

Comment:
I can't figure out how to trigger the 'Sync' error here. Are we sure there should be an issue with `AssertUnwindSafe`?

Reply:
why would sync trigger here? Also all objects are by default Send and Sync unles they contain something that is not Send/Sync, making Foo and i32 Send/Sync.

Reply:
It was an example used in one of the Issues. I was under the impression this should trigger an error? Should it not?

Reply:
not for the ysnc trait, but for the AssertUnwindSafe trati

Reply:
Ohh, so I should also check for this one?